core/math/rect2.h

```diff
@@ -285,13 +285,15 @@ struct [[nodiscard]] Rect2 {
 		return Rect2(position.round(), size.round());
 	}
 
-	Vector2 get_support(const Vector2 &p_normal) const {
-		Vector2 half_extents = size * 0.5f;
-		Vector2 ofs = position + half_extents;
-		return Vector2(
-					   (p_normal.x > 0) ? -half_extents.x : half_extents.x,
-					   (p_normal.y > 0) ? -half_extents.y : half_extents.y) +
-				ofs;
+	Vector2 get_support(const Vector2 &p_direction) const {
+		Vector2 support = position;
+		if (p_direction.x > 0.0f) {
+			support.x += size.x;
+		}
+		if (p_direction.y > 0.0f) {
+			support.y += size.y;
+		}
```

Comment:
This flips the behaviour of get_support. It seems like the older version wrongly returns the closest point along the direction instead of the furthest. Is `get_support()` used anywhere?

Reply:
Looks like it is only used for DirectionLight2D shadows. So we just need to fix those cases with this PR

https://github.com/godotengine/godot/blob/da5f39889f155658cef7f7ec3cc1abb94e17d815/drivers/gles3/rasterizer_canvas_gles3.cpp#L1737

https://github.com/godotengine/godot/blob/da5f39889f155658cef7f7ec3cc1abb94e17d815/servers/rendering/renderer_rd/renderer_canvas_render_rd.cpp#L1826

Reply:
Oh yes, you're correct. I didn't try running the new unit tests on the old version, I should've done this. But yeah the old function was completely backwards so I'll fix the places using this method.